controller/api/destination/server_test.go

```diff
@@ -970,3 +607,55 @@ status:
 		}
 	})
 }
+
+func assertSingleProfile(t *testing.T, updates []*pb.DestinationProfile) *pb.DestinationProfile {
+	t.Helper()
+	// Under normal conditions the creation of resources by the fake API will
+	// generate notifications that are discarded after the stream.Cancel() call,
+	// but very rarely those notifications might come after, in which case we'll
+	// get a second update.
+	if len(updates) != 1 {
+		t.Fatalf("Expected 1 profile update but got %d: %v", len(updates), updates)
+	}
+	return updates[0]
+}
+
+func assertSingleUpdate(t *testing.T, updates []*pb.Update) *pb.Update {
+	t.Helper()
+	// Under normal conditions the creation of resources by the fake API will
+	// generate notifications that are discarded after the stream.Cancel() call,
+	// but very rarely those notifications might come after, in which case we'll
+	// get a second update.
+	if len(updates) == 0 || len(updates) > 2 {
+		t.Fatalf("Expected 1 or 2 updates but got %d: %v", len(updates), updates)
+	}
```

Comment:
Isn't the condition equivalent to just `len(updates) != 1`? Also the error message currently doesn't match the condition.

Reply:
No, it's equivalent to `len(updates) != 1 && len(updates) != 2` -- this was in the pre-existing test.

Reply:
I don't actually know whether this logic is necessary or not but I was trying not to get into the `Get` behavior.

Reply:
Ah my bad